src/core/public/ui_settings/ui_settings_client.ts

```diff
@@ -173,6 +178,28 @@ You can use \`IUiSettingsClient.get("${key}", defaultValue)\`, which will just r
     return this.updateErrors$.asObservable();
   }
 
+  private getBrowserStoredSettings() {
+    const uiSettingsJSON = window.localStorage.getItem('uiSettings') || '{}';
+    try {
+      return JSON.parse(uiSettingsJSON);
+    } catch (error) {
+      this.updateErrors$.next(error);
+    }
+    return {};
+  }
+
+  private setBrowserStoredSettings(key: string, newVal: any) {
+    const oldSettings = this.getBrowserStoredSettings();
+    const newSettings = cloneDeep(oldSettings);
+    if (newVal === null) {
+      delete newSettings[key];
+    } else {
+      newSettings[key] = { userValue: newVal };
+    }
+    window.localStorage.setItem(`uiSettings`, JSON.stringify(newSettings));
+    return { settings: newSettings };
```

Comment:
Why not just return `newSettings` like `getBrowserStoredSettings`?

Reply:
Just noticed this probably just matches the return for `this.api.batchSet`

Reply:
Yeah - this was not really the ideal shape, but I did it this way to avoid some duplication. But I'm definitely open for ways to clean this up more and make it more readable and less weird.